go/vt/schemadiff/column.go

```diff
@@ -181,19 +219,228 @@ func (c *ColumnDefinitionEntity) ColumnDiff(
+	return isGenerated
+}
+
+// IsNullable returns true when this column is NULLable
+func (c *ColumnDefinitionEntity) IsNullable() bool {
+	if c.inPK {
+		return false
+	}
+	return c.ColumnDefinition.Type.Options.Null == nil || *c.ColumnDefinition.Type.Options.Null
+}
+
+// IsDefaultNull returns true when this column has DEFAULT NULL
+func (c *ColumnDefinitionEntity) IsDefaultNull() bool {
+	if !c.IsNullable() {
+		return false
+	}
+	_, ok := c.ColumnDefinition.Type.Options.Default.(*sqlparser.NullVal)
+	return ok
+}
+
+// IsDefaultNull returns true when this column has DEFAULT NULL
+func (c *ColumnDefinitionEntity) HasDefault() bool {
+	if c.ColumnDefinition.Type.Options.Default == nil {
+		return false
+	}
+	if c.IsDefaultNull() {
+		return true
+	}
+	return true
+}
+
+// IsAutoIncrement returns true when this column is AUTO_INCREMENT
+func (c *ColumnDefinitionEntity) IsAutoIncrement() bool {
+	return c.ColumnDefinition.Type.Options.Autoincrement
+}
+
+// IsUnsigned returns true when this column is UNSIGNED
+func (c *ColumnDefinitionEntity) IsUnsigned() bool {
+	return c.ColumnDefinition.Type.Unsigned
+}
+
+// IsNumeric returns true when this column is a numeric type
+func (c *ColumnDefinitionEntity) IsIntegralType() bool {
+	return IsIntegralType(c.Type())
+}
+
+// IsFloatingPointType returns true when this column is a floating point type
+func (c *ColumnDefinitionEntity) IsFloatingPointType() bool {
+	return IsFloatingPointType(c.Type())
+}
+
+// IsDecimalType returns true when this column is a decimal type
+func (c *ColumnDefinitionEntity) IsDecimalType() bool {
+	return IsDecimalType(c.Type())
+}
+
+// HasBlobTypeStorage returns true when this column is a text/blob type
+func (c *ColumnDefinitionEntity) HasBlobTypeStorage() bool {
+	return BlobTypeStorage(c.Type()) != 0
+}
+
+// Charset returns the column's charset
+func (c *ColumnDefinitionEntity) Charset() string {
+	return c.ColumnDefinition.Type.Charset.Name
+}
+
+// Collate returns the column's collation
+func (c *ColumnDefinitionEntity) Collate() string {
+	return c.ColumnDefinition.Type.Options.Collate
+}
+
+func (c *ColumnDefinitionEntity) EnumValues() []string {
+	return c.ColumnDefinition.Type.EnumValues
+}
+
+func (c *ColumnDefinitionEntity) HasEnumValues() bool {
+	return len(c.EnumValues()) > 0
+}
+
+// EnumValuesOrdinals returns a map of enum values to their ordinals
+func (c *ColumnDefinitionEntity) EnumValuesOrdinals() map[string]int {
+	m := make(map[string]int, len(c.ColumnDefinition.Type.EnumValues))
+	for i, enumValue := range c.ColumnDefinition.Type.EnumValues {
+		m[enumValue] = i
+	}
+	return m
+}
+
+// EnumOrdinalValues returns a map of enum ordinals to their values
+func (c *ColumnDefinitionEntity) EnumOrdinalValues() map[int]string {
+	m := make(map[int]string, len(c.ColumnDefinition.Type.EnumValues))
+	for i, enumValue := range c.ColumnDefinition.Type.EnumValues {
+		// SET and ENUM values are 1 indexed.
+		m[i+1] = enumValue
```

Comment:
This doesn't seem symmetric with `EnumValuesOrdinals`?

Reply:
Nice catch. Fixed `EnumValuesOrdinals`, which is as yet unused.

Reply:
Added tests